test/SlowTests/Issues/RavenDB-15497.cs

```diff
@@ -17,39 +18,44 @@ public RavenDB_15497(ITestOutputHelper output) : base(output)
         }
 
         [Fact]
-        public void WaitForIndexesAfterSaveChangesCanExitWhenThrowOnTimeoutIsFalse()
+        public async Task WaitForIndexesAfterSaveChangesCanExitWhenThrowOnTimeoutIsFalse()
         {
             using (var store = GetDocumentStore())
             {
                 var index = new Index();
-                index.Execute(store);
-                store.Maintenance.Send(new DisableIndexOperation(index.IndexName));
-
-                var indexStats = store.Maintenance.Send(new GetIndexStatisticsOperation(index.IndexName));
-                Assert.Equal(IndexState.Disabled, indexStats.State);
-                Assert.Equal(IndexRunningStatus.Disabled, indexStats.Status);
+                await index.ExecuteAsync(store);
 
                 using (var session = store.OpenSession())
                 {
                     session.Store(new User
                     {
-                        Name = "user1"
+                        Name = "user1",
+                        Count = 3
                     });
                     session.Advanced.WaitForIndexesAfterSaveChanges(timeout: TimeSpan.FromSeconds(3), throwOnTimeout: false);
                     session.SaveChanges();
                 }
 
+                Indexes.WaitForIndexing(store);
+
+                const int timeoutInMs = 2_000;
+                (await GetDatabase(store.Database)).IndexStore.GetIndex(index.IndexName).ForTestingPurposesOnly().BeforeStartingNewBatch = () =>
+                {
+                    Thread.Sleep(timeoutInMs + 100);
```

Comment:
instead of doing this, you can pause the indexing? Would like to avoid unnecessary thread.sleep